libconcurrent/concurrent/hsynch.c

```diff
@@ -52,6 +54,8 @@ RetVal HSynchApplyOp(HSynchStruct *l, HSynchThreadState *st_thread, RetVal (*sfu
         synchNonTSOFence();
         p->locked = false;
         p = tmp_next;
+        if (tmp_next->next == NULL && synchGetMachineModel() != INTEL_X86_MACHINE)
```

Comment:
Comments related to optimisations helps the reviewer (and code extensibility)

Reply:
Just an optimization here for newest Intel machines.

Reply:
In general I prefer to have comments in the code for parts no so straightforward such as this optimization.

Reply:
Check the e7df25c commit.